app/src/main/java/net/gsantner/markor/activity/SettingsActivity.java

```diff
@@ -206,6 +207,14 @@ protected void onPreferenceChanged(SharedPreferences prefs, String key) {
                 boolean extraLaunchersEnabled = prefs.getBoolean(key, false);
                 ActivityUtils au = new ActivityUtils(getActivity());
                 au.applySpecialLaunchersVisibility(extraLaunchersEnabled);
+            } else if (eq(key, R.string.pref_key__default_encryption_password)
+                    && Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+                final String pw = prefs.getString(key, null);
+                final SharedPreferences.Editor edit = prefs.edit();
+                edit.remove(key);
+                edit.commit();
+                final PasswordStore store = new PasswordStore(this.getActivity());
+                store.storeKey(pw);
```

Comment:
@opensource21 
Is it  also working if you  have it like this?
```
new PasswordStore(this.getActivity()).storeKey(prefs.getString(key, null));
prefs.edit().remove(key).commit();
```

Reply:
It's tricky, because we come into an loop, during change we make new change, which triggers a new notification.

Reply:
you can do:
```
if (!TextUtils.isEmpty(prefs.getString(key, null))) {
    new PasswordStore(this.getActivity()).storeKey(prefs.getString(key, null));
    prefs.edit().remove(key).commit();
}
```

Reply:
Solved it with an improved password-store. Now you see *** if and only if a password is set. 
You can delete the password by setting it to an empty string.
Only drawback *** is not a valid password, but I think this is acceptable.